src/controllers/DynamicCategoryFacetQueryController.ts

```diff
@@ -55,7 +56,29 @@ export class DynamicCategoryFacetQueryController {
     };
   }
 
-  public get currentValues(): IFacetRequestValue[] {
+  public executeIsolatedQuery(): Promise<IQueryResults> {
+    const query = this.facet.queryController.getLastQuery();
+    // Specifying a numberOfResults of 0 will not log the query as a full fledged query in the API
+    // it will also alleviate the load on the index
+    query.numberOfResults = 0;
+
+    const previousFacetRequestIndex = findIndex(query.facets, { facetId: this.facet.options.id });
+    if (previousFacetRequestIndex !== -1) {
+      query.facets[previousFacetRequestIndex] = this.facetRequest;
+    } else if (query.facets) {
+      query.facets.push(this.facetRequest);
+    } else {
+      query.facets = [this.facetRequest];
+    }
+
+    return this.facet.queryController.getEndpoint().search(query);
+  }
+
+  private get currentValues(): IFacetRequestValue[] {
+    // TODO: remove when API has fixed currentValue/numberOfValues issue
+    if (!this.facet.values.hasSelectedValue && this.numberOfValuesToRequest > this.facet.options.numberOfValues) {
```

Comment:
If we passed currentValues API-side in order to fetch more values, it would return an error. I will follow up the error and remove the check when it won't be necessary.